posthog/auth.py

```diff
@@ -68,8 +69,13 @@ def authenticate(cls, request: Union[HttpRequest, Request]) -> Optional[Tuple[An
             )
         except PersonalAPIKey.DoesNotExist:
             raise AuthenticationFailed(detail=f"Personal API key found in request {source} is invalid.")
-        personal_api_key_object.last_used_at = timezone.now()
-        personal_api_key_object.save()
+
+        now = timezone.now()
+        if personal_api_key_object.last_used_at is None or now - personal_api_key_object.last_used_at > timedelta(
+            hours=12
```

Comment:
Great idea, it's totally dumb to update this every single time (I wrote this soon after I started at PostHog LOL). From a UX perspective however, I propose making the min delta 1 hour and updating the UI to only show the hour instead of a full datetime, so that users are not fooled – false precision in an auth feature looks bad.